### ISSUE TITLE
iOS app from PGB: 400: An update check must include a binary version that conforms to the semver standard (e.g. '1.0.0').

### ISSUE THREAD

**Issue by ef33zy** [Microsoft/cordova-plugin-code-push#187]
Hi,

I am using code-push plugin for my app built using PhoneGap Build.

Firstly, the plugin is completely dependent on using cli, which defeats the PGB purpose... nonetheless, I've managed to get the pushing of updates working on Android.

However, when I'm debugging my iOS, I see the below error:
``
[CodePush]  400: An update check must include a binary version that conforms to the semver standard (e.g. '1.0.0'). The binary version is normally inferred from the App Store/Play Store version configured with your app.
``

In my app's `config.xml` file, I have defined the version in the widget as below:
`<widget version = "2.1.0">`

I also have the below config-file element defined.
```
<config-file platform="ios" parent="CFBundleShortVersionString">
<string>210</string>
</config-file>
```

Is the plugin not able to recognize either of these?
Is there something I am missing?
I'm not quite sure how else to define the `binary version` for my app.

I have also raised the same issue on the  code-push repository [here](https://github.com/Microsoft/code-push/issues/347).

Your help & responses are highly appreciated.

**Comment by matrosov-nikita** [comment]
> I also have the below config-file element defined.

There is probably no sense in having this, because Cordova already uses `widget`'s `version` attribute [to populate `CFBundleShortVersionString`](https://github.com/apache/cordova-ios/blob/b8ae88d822135a149ff1ea7b6e5feb045a2220eb/bin/templates/scripts/cordova/lib/prepare.js#L201)

Anyway, it looks like the version sent by client app to server appears 'invalid'. There are two sources where it might come from:

1. `Info.plist` packed with app bundle
2. `currentPackage.json` from application data directory (usually something like `/data/data/<BUNDLE_ID>/files/codepush/currentPackage.json`)

Could you please check these files to see what version is stored there? (you can check [this SO post](http://stackoverflow.com/questions/25937785/how-to-download-app-data-in-xcode-6) for instructions)

Also it'd be great if you could capture the request that app sends to code push server when it checks available updates, to see what version it reports

**Comment by sergey-akhalkov** [comment]
@ef33zy, do you still experiencing the issue? Please let us know if you need any help.

**Comment by ef33zy** [comment]
Hi @matrosov-nikita, @sergey-akhalkov apologies for not posting earlier.

Since I am building with PhoneGap Build, I removed the below section from my `config.xml` file, and I didn't see the problem anymore.

```
<config-file platform="ios" parent="CFBundleShortVersionString">
<string>210</string>
</config-file>
```

I have also closed out the [issue #347](https://github.com/Microsoft/code-push/issues/347) I raised in the main code-push repository.
